test/s3select_test.cpp

```diff
@@ -280,25 +280,25 @@ TEST(TestS3SElect, floatnan_compare_operator)
   ASSERT_EQ( d >= c10, false );
   ASSERT_EQ( d < a10, false );
   ASSERT_EQ( d <= b11, false );
-  ASSERT_EQ( d != a10, true );
-  ASSERT_EQ( d != e, true );
-  ASSERT_EQ( d == a10, false );
```

Comment:
https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN   (compare with NAN)
for (NaN ≠ x) its always TRUE